### PR TITLE
Projects Enhancements

### PR DESCRIPTION
**Pull request by Nitish145**
### Describe the changes you have made in this PR -
- User Projects / Featured Projects (public only)  are now available without authentication as they should be..
- User Projects /Featured Projects (private) would still require authentication.
- Add user_favourites `/api/v1/users/:id/favourites` endpoint.
- Add / Update specs for the above mentioned changes.

### Screenshots of the changes (If any) -

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

**Comment by Nitish145** [comment]
@tachyons please check!!

File: app/controllers/api/v1/projects_controller.rb [hunk 1]
```diff
@@ -113,10 +115,25 @@ def load_index_projects
     end
 
     def load_user_projects
-      @projects = if @current_user.id == params[:id].to_i
+      # if user is not authenticated or authenticated as some other user
+      # return only user's public projects else all
+      @projects = if @current_user.nil? || @current_user.id != params[:id].to_i
```
Comment (tachyons) [inline app/controllers/api/v1/projects_controller.rb:120]: `@current_user.id != params[:id].to_i` why are you excluding this? here both triggers same db query

Comment (Nitish145) [reply app/controllers/api/v1/projects_controller.rb:120]: What if some user fetches some other user's projects? Shouldn't we only return public projects then? @tachyons

Comment (tachyons) [reply app/controllers/api/v1/projects_controller.rb:120]: Got it, thanks

**Comment by tachyons** [comment]
Resolve conflicts

**Comment by Nitish145** [comment]
> Resolve conflicts

Done @tachyons

**Comment by tachyons** [comment]
Refer https://github.com/CircuitVerse/CircuitVerse/commit/faea53637ed922aaab8fce39675313924b71b8b8 @current_user won't work anymore

**Comment by Nitish145** [comment]
> Refer https://github.com/CircuitVerse/CircuitVerse/commit/faea53637ed922aaab8fce39675313924b71b8b8 @current_user won't work anymore 

Yeah! I did will be pushing the required changes..

**Pull request merged**